test/EndToEndTests/Services/ServiceWrapperApp/ServiceWrapperApp.cs

```diff
@@ -0,0 +1,319 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ServiceWrapperApp.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace ServiceWrapperApp
+{
+    using System;
+    using System.Linq;
+    using Microsoft.Test.OData.Framework.Server;
+    using Microsoft.Test.OData.Services.TestServices;
+    using ServiceCommand = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceCommands;
+    using ServiceDescriptorType = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceDescriptorType;
+    using ServiceType = Microsoft.Test.OData.Tests.Client.IPCCommandMap.ServiceType;
+
+    /// <summary>
+    /// Commandline tool to launch the test services for OData Client E2E tests. This tool must be run in admin mode to open ports.
+    /// </summary>
+    class ServiceWrapperApp
+    {
+        private static ServiceDescriptor serviceDescriptor;
+        private static IServiceWrapper serviceWrapper;
+        private static bool isServiceRunning = false;
+
+        /// <summary>
+        /// App entry.
+        /// </summary>
+        /// <param name="args">Arguments passed into commandline.</param>
+        static int Main(string[] args)
+        {
+            ServiceCommand serviceCommand;
+            ServiceDescriptorType serviceDescriptorType;
+            ServiceType serviceType;
+            string commandText;
+
+            if (args.Contains("h") || args.Contains("help") || args.Contains("?"))
+            {
+                PrintHelp();

```

Comment:
>PrintHelp(); [](start = 16, length = 12)

only printing help when they explicitly ask for it? wouldn't we like to output the help if they get any kind of issue with their inputs?

Reply:
Done.